src/v/pandaproxy/schema_registry/client_fetch_batch_reader.cc

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "pandaproxy/schema_registry/client_fetch_batch_reader.h"
+
+#include "kafka/protocol/exceptions.h"
+#include "kafka/protocol/kafka_batch_adapter.h"
+#include "pandaproxy/logger.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/std-coroutine.hh>
+
+namespace pandaproxy::schema_registry {
+
+class client_fetcher final : public model::record_batch_reader::impl {
```

Comment:
@graphcareful can you take a look at this?

Reply:
Left my thoughts on this in another thread in this PR